modules/utxo-coredao/src/opReturn.ts

```diff
@@ -0,0 +1,255 @@
+// Source: https://docs.coredao.org/docs/Learn/products/btc-staking/design
+export const CORE_DAO_TESTNET_CHAIN_ID = Buffer.from('045b', 'hex');
+export const CORE_DAO_MAINNET_CHAIN_ID = Buffer.from('045c', 'hex');
+export const CORE_DAO_SATOSHI_PLUS_IDENTIFIER = Buffer.from('5341542b', 'hex');
+// https://github.com/bitcoin/bitcoin/blob/5961b23898ee7c0af2626c46d5d70e80136578d3/src/script/script.h#L47
+const OP_RETURN_IDENTIFIER = Buffer.from('6a', 'hex');
+const OP_PUSHDATA1_IDENTIFIER = Buffer.from('4c', 'hex');
+const OP_PUSHDATA2_IDENTIFIER = Buffer.from('4d', 'hex');
+const OP_PUSHDATA4_IDENTIFIER = Buffer.from('4e', 'hex');
+
+export function encodeTimelock(timelock: number): Buffer {
+  const buff = Buffer.alloc(4);
+  buff.writeUInt32LE(timelock);
+  return buff;
+}
+
+export function decodeTimelock(buffer: Buffer): number {
+  if (buffer.length !== 4) {
+    throw new Error('Invalid timelock buffer length');
+  }
+  return buffer.readUInt32LE();
+}
+
+export function encodeOpReturnLength(length: number): Buffer {
```

Comment:
I'm almost positive that

```
bitcoin.script.compile([OPS.OP_RETURN, buffer]);
```

does the right thing and performs these calculations for you - please take a look


see here for instance: https://github.com/bitcoinjs/bitcoinjs-lib/blob/32e08aa57f6a023e995d8c4f0c9fbdc5f11d1fa0/ts_src/payments/embed.ts#L38

you could use `payments.p2data()` as well maybe

Reply:
in fact you should be able to use `payments.p2data()` to encode and decode `OP_RETURN` outputs